tracing/tracing.go

```diff
@@ -0,0 +1,36 @@
+package tracing
+
+import (
+	propjaeger "go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/trace/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// InitTracer creates an OTel TracerProvider that exports the traces to a Jaeger collector.
+func InitTracer(serviceName, collectorEndpoint string, samplingFraction float64) (tp trace.TracerProvider, closer func(), err error) {
+	disabled := collectorEndpoint == ""
+
+	tp, closer, err = jaeger.NewExportPipeline(
+		jaeger.WithCollectorEndpoint(collectorEndpoint),
+		jaeger.WithProcess(jaeger.Process{
+			ServiceName: serviceName,
+		}),
+		jaeger.WithSDK(&sdktrace.Config{DefaultSampler: sdktrace.TraceIDRatioBased(samplingFraction)}),
+		jaeger.WithDisabled(disabled),
+	)
+	if err != nil {
+		return
```

Comment:
wooo wooo, let's hold on here. Have you see our magic Thanos style guide? 😉 https://thanos.io/v0.18/contributing/coding-style-guide.md/#use-named-return-parameters-carefully